### ISSUE TITLE
Lowercase hex (e.g. #fff) do not convert properly

### ISSUE THREAD

**Issue by Makaze** [py-pdf/fpdf2#1225]
Cannot use lowercase letters in hex, e.g. `#fff`.

**Error details**
If an exception is raised, it is very important that you provide the full error message.
Otherwise members of the fpdf2 community won't be able to help you with your problem.

**Minimal code**
Please include some minimal Python code reproducing your issue:
```python
from fpdf import FPDF

pdf = FPDF()
pdf.set_text_color("#fff")  # Does not convert
pdf.set_text_color("#FFF")  # Does convert
```
If you don't know how to build a minimal reproducible example, please check this tutorial: https://stackoverflow.com/help/minimal-reproducible-example

**Environment**
Please provide the following information:
* **Operating System**: Linux
* **Python version**: Python 3.10.12
* **`fpdf2` version used**: fpdf2-2.7.9

**Comment by gmischler** [comment]
Welcome to fpdf2, @Makaze !

Can you please explain/demonstrate the exact error or incorrect result you get?

`pdf.set_text_color()` uses `color_from_hex_string()`, which in turn uses `int(c, base=16)` to convert the color component substrings. And nowhere in there can I see a reason why uppercase or lowercase hex strings (or even a mix thereof) would result in any difference of the output.

**Comment by Makaze** [comment]
I am thoroughly embarrassed.

The issue: Using a hex code (uppercase or lowercase) causes code to fail CS50's shirtificate problem. However, there is no issue with the file manually.

 I believe the issue stems from the some kind of linting process: My LSP (pyright) tells me that str literals are not acceptable arguments to that function. I am not sure if fixing the type hinting/docs would solve this problem, but there is no actual issue in the execution.

The only valuable thing I added: Your docs state it accepts uppercase, and you did not have a test case for lowercase hex.

In short, I jumped the gun entirely. I will close and rescind my PR. Thank you for your time!

**Comment by gmischler** [comment]
> The issue: Using a hex code (uppercase or lowercase) causes code to fail CS50's shirtificate problem.

Huh, bummer!

Kind of curious, though: Did their test procedures offer any kind of specific feedback?
It's surprising that pyright complains about the argument type, since none of the involved methods appear to have type hints. Are taking their hints from the docstring?
The list of allowable types there is indeed incomplete and doesn't include strings for the first parameter.

Btw: You could keep your PR going just with the documentation and test improvements if you want to, so your effort doesn't end up entierely wasted.

**Comment by Makaze** [comment]
I did track it down about an hour ago. The check50 server was using an older version.

The error occurred due to using version `fpdf2==2.7.6`. The error generated at that version was:
```
...
    pdf.output("shirtificate.pdf")
  File "/home/makaze/.local/lib/python3.10/site-packages/fpdf/fpdf.py", line 4916, in output
    self.footer()
  File "/home/makaze/Documents/Harvard/CS50P/shirtificate/shirtificate.py", line 27, in footer
    self.set_text_color("#FFF")
  File "/home/makaze/.local/lib/python3.10/site-packages/fpdf/fpdf.py", line 1037, in set_text_color
    self.text_color = _convert_to_drawing_color(r, g, b)
  File "/home/makaze/.local/lib/python3.10/site-packages/fpdf/fpdf.py", line 4943, in _convert_to_drawing_color
    r, g, b = r
    
ValueError: too many values to unpack (expected 3)
```
